obspy/taup/tau.py

```diff
@@ -31,6 +58,131 @@ def __str__(self):
     def __repr__(self):
         return "[%s]" % (", ".join([repr(_i) for _i in self]))
 
+    def plot(self, plot_type="spherical", plot_all=True, ax=None, show=True):
+        """
+        Plot the ray paths if any have been calculated.
+
+        :param plot_type: Either ``"spherical"`` or ``"cartesian"``.
+            A spherical plot is always global whereas a cartesian one can
+            also be local.
+        :type plot_type: str
+        :param plot_all: By default only rays that match the requested
+            distance are plotted and rays arriving at 360 - x degrees are
+            not shown.
+        :type plot_all: bool
+        :param ax: Axes to plot to. If not given, a new figure with an axes
+            will be created. Must be a polar axes for the spherical plot and
+            a regular one for the cartesian plot.
+        :type ax: :class:`matplotlib.axes.Axes`
+        :param show: Show the plot.
+        :type show: bool
+
+        :returns: The (possibly created) axes instance.
+        :rtype: :class:`matplotlib.axes.Axes`
+        """
+        arrivals = []
+        for _i in self:
+            if _i.path is None:
+                continue
+            dist = _i.purist_distance % 360.0
+            if abs(dist - self.distance) / dist > 1E-5:
+                if plot_all is False:
+                    continue
+                # Mirror on axis.
```

Comment:
I also thought about this and am not really sure as we'd then end up with negative distances which might confuse even more people.